smart_contracts.rst

```diff
@@ -40,6 +40,128 @@ Project Requirements
 - Determine if and how different versions of the smart contracts should interoperate.
 - Determine if and how channels should be upgraded.
 
+Data structures
+===============
+
+.. _balance-proof-message:
+
+Balance Proof
+-------------
+
+Data required by the smart contracts to update the payment channel end of the participant that signed the balance proof.
+The signature must be valid and is defined as:
+
+::
+
+    ecdsa_recoverable(privkey, keccak256(balance_hash || nonce || additional_hash || channel_identifier || token_network_address || chain_id)
+
+Fields
+^^^^^^
+
++------------------------+------------+--------------------------------------------------------------------------------+
+| Field Name             | Field Type |  Description                                                                   |
++========================+============+================================================================================+
+|  balance_hash          | bytes32    | Balance data hash                                                              |
++------------------------+------------+--------------------------------------------------------------------------------+
+|  nonce                 | uint64     | Strictly monotonic value used to order transfers. The nonce starts at 1        |
```

Comment:
Also, `nonce` is now `uint256` - https://github.com/raiden-network/raiden-contracts/blob/dd0b258d79622211a8605012038d39ef710b0f67/raiden_contracts/contracts/TokenNetwork.sol#L83, we need to also fix this in the spec.